liberty-maven-plugin/src/it/dev-it/src/test/java/net/wasdev/wlp/test/dev/it/DevTest.java

```diff
@@ -210,6 +210,40 @@ public void resolveDependencyTest() throws Exception {
       assertTrue(systemHealthTarget.exists());
    }
 
+   // TBD @Test
+   public void generateFeatureTest() throws Exception {
+
+      final String GENERATED_FEATURES_FILE_NAME = "liberty-plugin-added-features.xml";
+      final String SERVER_XML_COMMENT = "Plugin has generated Liberty features"; // the explanation added to server.xml
+      // TBD final String NEW_FILE_INFO_MESSAGE = "Some message"; // the explanation added to the generated features file
+
+      assertTrue(verifyLogMessageExists("Liberty is running in dev mode.", 10000));
+      assertFalse(verifyLogMessageExists("batch-1.0", 10000));
+
+      File newFeatureFile = new File(tempProj, "/src/main/liberty/config/configDropins/overrides/"+GENERATED_FEATURES_FILE_NAME);
+      File newTargetFeatureFile = new File(targetDir, "/liberty/wlp/usr/servers/defaultServer/configDropins/overrides/"+GENERATED_FEATURES_FILE_NAME);
+      File serverXmlFile = new File(tempProj, "/src/main/liberty/config/server.xml");
+      assertTrue(serverXmlFile.exists());
+
+      // Copy a Java file into place to create the HelloBatch class
+      File helloBatchRes = new File("../resources/HelloBatch.java");
+      assertTrue(helloBatchRes.exists());
+      File helloBatchSrc = new File(tempProj, "/src/main/java/com/demo/HelloBatch.java");
+      FileUtils.copyFile(helloBatchRes, helloBatchSrc);
+      assertTrue(helloBatchSrc.exists());
+
+      // Dev mode should now compile the new Java file...
+      Thread.sleep(15000);
+      // ... and run the proper mojo.
+      assertTrue(verifyLogMessageExists("Running liberty:generate-features", 10000)); // mojo ran
+      assertTrue(newFeatureFile.exists()); // mojo created file
```

Comment:
Could there be a timing issue here, since line 238 will return when "Running liberty:generate-features" is detected but the mojo might not be finished?

Reply:
Definitely a race condition but now we have verifyFileExists() we can use it here.